notebooks/config.ipynb

```diff
@@ -0,0 +1,1046 @@
+      "text/plain": [
+       "{'greeting': 'Hello', 'recipient': 'World'}"
+      ]
+     },
+     "execution_count": 11,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "config.realize(\n",
+    "    input_config=config1,\n",
+    "    output_file=Path('tmp/config1.yaml')\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "fd1823c4-6c3d-4d4a-a614-4d1238588bdd",
+   "metadata": {},
+   "source": [
+    "The function returns a `dict` version of the config, and the file is written in the YAML format as indicated by the file extension.\n",
+    "<!--cell 22-->"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 12,
+   "id": "4f237a73-da83-4632-990f-644632b15cd9",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "greeting: Hello\n",
+      "recipient: World\n"
+     ]
+    }
+   ],
+   "source": [
+    "%%bash\n",
+    "cat tmp/config1.yaml"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "902c4863-e772-49c6-88f2-31e1ab76b418",
+   "metadata": {},
+   "source": [
+    "Input and output formats are not required to match. This can be used to convert a config from one format to another.\n",
+    "<!--cell 24-->"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 13,
+   "id": "b8854dc6-9dd2-4843-99e4-278b116b9767",
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "{'greeting': 'Hello', 'recipient': 'World'}"
+      ]
+     },
+     "execution_count": 13,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "config.realize(\n",
+    "    input_config='fixtures/config/get-config.yaml',\n",
+    "    input_format='yaml',\n",
+    "    output_file='tmp/realize-config.sh',\n",
+    "    output_format='sh'\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "4032ab38-f6c7-45d1-bff7-f70d23832f26",
+   "metadata": {},
+   "source": [
+    "Here a Shell config is created from a YAML config.\n",
+    "<!--cell 26-->"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 14,
+   "id": "0a59a3e8-27b5-4daa-a924-941aceaad157",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "greeting=Hello\n",
+      "recipient=World\n"
+     ]
+    }
+   ],
+   "source": [
+    "%%bash\n",
+    "cat tmp/realize-config.sh"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "ef0136e4-f549-467f-9341-f77f84738bb0",
+   "metadata": {},
+   "source": [
+    "### Updating configs\n",
+    "\n",
+    "Configs can be updated by providing a second config with the `update_config` parameter. If the update config contains keys that match the base config, they will be overwritten. If the base config contains Jinja2 expressions, like the one below, they will be rendered using the values in the update config wherever possible.\n",
```

Comment:
```suggestion
    "Configs can be updated by providing a second config with the `update_config` parameter. If the update config contains keys that match the base config, the base config values for those keys will be overwritten. Once updated,  if the config contains Jinja2 expressions, like the one below, they will be rendered in the config wherever possible.\n",
```

I believe we should be able to provide an update config with Jinja2 expressions and they'd be rendered as well. I added some suggested changes here that reflect that.

In other words, once the updated values are applied to the base config, the new total config is dereferenced -- as many Jinja2 expressions as possible are rendered.